execution/scan/matrix_selector.go

```diff
@@ -32,12 +32,13 @@ type matrixScanner struct {
 }
 
 type matrixSelector struct {
-	funcExpr *parser.Call
-	storage  engstore.SeriesSelector
-	call     function.FunctionCall
-	scanners []matrixScanner
-	series   []labels.Labels
-	once     sync.Once
+	funcExpr       *parser.Call
+	storage        engstore.SeriesSelector
+	call           function.FunctionCall
+	scanners       []matrixScanner
+	metricAppeared []int64
```

Comment:
Should we put this in the scanner itself? Also does it denote the time when the metric first appeared?